### PR TITLE
[feat] Stormtrade-Perps: add storm trade oi adapter via api

### PR DESCRIPTION
**Pull request by Shaileshkhote**
## Summary
- Adds a Storm Trade open interest adapter for TON.
- Uses Storm Trade's live markets API to report current total, long, and short OI.

## Changes
- Added `open-interest/stormtrade-oi.ts`.
- Aggregates `openInterestLong` and `openInterestShort` from `https://api.storm.tg/api/markets`.
- Scales Storm Trade's 1e9-based USD notional values and excludes hidden markets.

## Methodology
- Reports current OI only because the Storm Trade markets API returns live snapshots, not historical OI.
- Calculates total OI as long OI plus short OI to avoid signed notional edge cases.

## Tests
- `pnpm test open-interest stormtrade-oi`

### CODE REVIEW

File: open-interest/stormtrade-oi.ts [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    if (market.config?.isHidden) continue;
+    longOpenInterestAtEnd += Number(market.amm?.openInterestLong) / USD_SCALE;
+    shortOpenInterestAtEnd += Number(market.amm?.openInterestShort) / USD_SCALE;
+  }
+
+  return {
+    openInterestAtEnd: longOpenInterestAtEnd + shortOpenInterestAtEnd,
+    longOpenInterestAtEnd,
+    shortOpenInterestAtEnd,
+  };
```
[the remark this answers is not in the document]
Comment (Shaileshkhote) [reply open-interest/stormtrade-oi.ts]: ignore, i already explained errors are better than wrong data

File: open-interest/stormtrade-oi.ts [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+const fetch = async () => {
+  const markets = await fetchURL(MARKETS_URL);
+
+  let longOpenInterestAtEnd = 0;
+  let shortOpenInterestAtEnd = 0;
+  for (const market of markets) {
+    if (market.config?.isHidden) continue;
+    longOpenInterestAtEnd += Number(market.amm?.openInterestLong) / USD_SCALE;
+    shortOpenInterestAtEnd += Number(market.amm?.openInterestShort) / USD_SCALE;
```
[the remark this answers is not in the document]
Comment (Shaileshkhote) [reply open-interest/stormtrade-oi.ts]: errors accepter rather than wrong data

File: open-interest/stormtrade-oi.ts [hunk 3]
```diff
@@ -0,0 +1,33 @@
+import { SimpleAdapter } from "../adapters/types";
+import { CHAIN } from "../helpers/chains";
+import fetchURL from "../utils/fetchURL";
+
+const MARKETS_URL = "https://api.storm.tg/api/markets";
+const USD_SCALE = 1e9;
+
+const fetch = async () => {
```
[the remark this answers is not in the document]
Comment (Shaileshkhote) [reply open-interest/stormtrade-oi.ts]: fixed in followup commit

**Comment by bheluga** [comment]
@Shaileshkhote thanks for the PR
